validation/superchain-version_test.go

```diff
@@ -13,12 +14,37 @@ import (
 	"github.com/ethereum-optimism/superchain-registry/validation/standard"
 	"github.com/stretchr/testify/require"
 
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/crypto"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/ethclient"
 )
 
+var (
+	// list of contracts to check for version/bytecode uniformity
+	// TODO - The list intentionally omits contracts which have immutables because the bytecode check needs
+	// to be enhanced to mask the immutable values in the bytecode (contracts related to fault proofs) so that any check against a hash
+	// of the bytecode can validate what's on-chain.
+	contractsToCheck = []string{
+		"L1CrossDomainMessengerProxy",
+		"L1ERC721BridgeProxy",
+		"L1StandardBridgeProxy",
+		"OptimismMintableERC20FactoryProxy",
+		"OptimismPortalProxy",
+		"SystemConfigProxy",
+		// "AnchorStateRegistryProxy",
+		// "DelayedWETHProxy",
+		// "DisputeGameFactoryProxy",
+		// "FaultDisputeGame",
+		// "MIPS",
+		"PermissionedDisputeGame",
+		"PreimageOracle",
+	}
```

Comment:
Longer term, a unified list of contracts to check the version and bytecode for makes sense. However, given that we are not yet tackling all the Fault Proof contracts I think it makes sense to keep two lists, or possibly better one "full" list, and then a list of `contractsToSkip` for the bytecode checks (with a comment and link to a github issue tracking the work that is still to be done).

Reply:
Ah, the `contractsToSkip` sounds like a good idea. will tweak.